cpp/src/parquet/file_writer.cc

```diff
@@ -361,7 +361,10 @@ std::unique_ptr<ParquetFileWriter> ParquetFileWriter::Open(
               key_value_metadata);
 }
 
-void WriteFileMetaData(const FileMetaData& file_metadata, ArrowOutputStream* sink) {
+void WriteFileMetaData(const FileMetaData& file_metadata, ArrowOutputStream* sink,
+                       const bool metafile) {
+  if (metafile) PARQUET_THROW_NOT_OK(sink->Write(PARQUET_MAGIC, 4));
```

Comment:
I'm wondering if there's ever a situation where we would _not_ want the magic number first. I would say to write it unconditionally and not add the extra function argument.

Reply:
I think the problem is that if it is a stand-alone file, this is needed. But this function is otherwise used to write the FileMetadata into a in-progress parquet file with actual data, and then the writing of this magic number is already handled by the writer one level above.

An alternative would be to have a `FileMetadata::WriteMetadataFile` to write stand-alone metadata files (as there was in a previous version of the PR), but let that use this WriteFileMetaData function to avoid duplication of the content (and in addition it can write the magic function).

Reply:
Ah, I see, I missed that. Yeah, I think adding a new API would be good rather than having a flag